projects/vanilla/src/app/configurators/metadata-configurator.component.ts

```diff
@@ -7,43 +7,162 @@ import { ComponentConfig, ConfiguratorContext } from "@sinequa/ngx-ui-builder";
+        <label class="form-check-label" for="filterable">Filterable</label>
+      </div>
+      <div class="form-check mb-2">
+        <input class="form-check-input" type="checkbox" id="excludable" [(ngModel)]="metadataToModify.excludable" (ngModelChangeDebounced)="configChanged()">
+        <label class="form-check-label" for="excludable">Excludable</label>
+      </div>
+      <div class="form-check mb-2">
+        <input class="form-check-input" type="checkbox" id="collapseRows" [(ngModel)]="metadataToModify.collapseRows" (ngModelChangeDebounced)="configChanged()">
+        <label class="form-check-label" for="collapseRows">Collapse rows</label>
+      </div>
+      <div class="form-check mb-2" *ngIf="isEntity">
+        <input class="form-check-input" type="checkbox" id="showEntityExtract" [(ngModel)]="metadataToModify.showEntityExtract" (ngModelChangeDebounced)="configChanged()">
+        <label class="form-check-label" for="showEntityExtract">Show entity extract</label>
+      </div>
+      <div class="mb-2" *ngIf="isEntity">
+        <label class="form-label" for="entityExtractMaxLines">Max lines for entity extract</label>
+        <input type="number" class="form-control" id="entityExtractMaxLines" [(ngModel)]="metadataToModify.entityExtractMaxLines">
+      </div>
+    </ng-container>
   </div>
 </div>
     `
 })
 export class MetadataConfiguratorComponent implements OnChanges {
   @Input() context: ConfiguratorContext;
-  @Input() metadata: MetadataConfig[];
+  @Input() metadata: string[];
 
+  modifiedMetadata: string;
+  metadataString: string;
+  metadataToModify: MetadataConfig;
+  stringToModify?: string;
+  newString: string;
   isEntity: boolean;
 
-  constructor(public appService: AppService) {}
+  constructor(public appService: AppService) { }
 
-  ngOnChanges(): void {
-    this.isEntity = this.appService.isEntity(this.config.item);
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.context && this.metadataToModify) {
+      this.metadataToModify = this.config.metadataConfig.find(m => m.field === this.metadataToModify.field);
+    }
   }
 
   get config(): ComponentConfig {
     return this.context.config;
   }
 
-  configChanged(isItem?: boolean) {
-    if (isItem) {
-      this.isEntity = this.appService.isEntity(this.config.item);
-    }
+  get metadataSelection(): string[] {
+    const stringItems = this.config.metadataConfig?.filter(c => !c.field) || [];
+    return this.metadata.concat(stringItems);
+  }
+
+ /** Update the whole config */
+  configChanged() {
     this.context.configChanged();
   }
+
+  /** Selecting a metadata to update */
+  selectedMetadata() {
+    this.metadataToModify = this.config.metadataConfig.find(m => m.field === this.modifiedMetadata);
+    this.stringToModify = this.metadataToModify ? undefined : this.modifiedMetadata;
+    this.newString = String(this.stringToModify);
+    this.isEntity = this.metadataToModify && this.appService.isEntity(this.metadataToModify.field);
```

Comment:
The find() is made on the config.metadataConfig, which actually cannot be undefined here as this method is only accessible when we select a metadata from the select which is only displayed when config.metadataConfig has items

Reply:
and if the find returns undefined, then in the isEntity assignment we won't go up to the metadataToModify.field instruction since we first check if metadataToModify is set